Prefabs/Interactions/Controllables/SharedResources/Scripts/RotationalJointDrive.cs

```diff
@@ -0,0 +1,224 @@
+            }
+
+            pseudoRotation = currentActualAngle + (circleDegrees * rotationMultiplier);
+            jointMotor.targetVelocity = (facade.MoveToTargetValue && !pseudoRotation.ApproxEquals(actualTargetAngle, targetValueReachedThreshold) ? facade.DriveSpeed : 0f) * CalculateDirectionMultiplier();
+            joint.motor = jointMotor;
+
+            if (facade.MoveToTargetValue && jointMotor.targetVelocity == 0f)
+            {
+                joint.transform.localRotation = Quaternion.Euler(joint.axis * actualTargetAngle);
+            }
+        }
+
+        /// <inheritdoc />
+        public override Vector3 CalculateDriveAxis(DriveAxis driveAxis)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return Vector3.zero;
+            }
+
+            Vector3 axisDirection = base.CalculateDriveAxis(driveAxis);
+            joint.axis = axisDirection * -1f;
+            return axisDirection;
+        }
+
+        /// <inheritdoc />
+        public override FloatRange CalculateDriveLimits(FloatRange newLimit)
+        {
+            return newLimit;
+        }
+
+        /// <inheritdoc />
+        public override void CalculateHingeLocation(Vector3 newHingeLocation)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return;
+            }
+
+            joint.anchor = newHingeLocation;
+            SetJointContainerPosition();
+        }
+
+        /// <inheritdoc />
+        public override void ProcessDriveSpeed(float driveSpeed, bool moveToTargetValue)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void SetTargetValue(float normalizedValue)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void ProcessAutoDrive(bool autoDrive)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return;
+            }
+
+            joint.useMotor = autoDrive;
+        }
+
+        /// <inheritdoc />
+        protected override float CalculateValue(DriveAxis givenAxis, FloatRange givenLimits)
+        {
+            if (!isActiveAndEnabled)
+            {
+                return 0f;
+            }
+
+            return Mathf.Clamp(pseudoRotation, givenLimits.minimum, givenLimits.maximum);
+        }
+
+        /// <inheritdoc />
+        protected override void InternalSetup()
+        {
+            jointMotor.force = joint.motor.force;
+            ProcessAutoDrive(facade.MoveToTargetValue);
+            jointRigidbody = joint.GetComponent<Rigidbody>();
+            CalculateHingeLocation(facade.HingeLocation);
+            base.InternalSetup();
+        }
+
+        /// <summary>
+        /// Sets the <see cref="jointContainer"/> position based on the joint anchor.
+        /// </summary>
+        protected virtual void SetJointContainerPosition()
+        {
+            //Disable any child rigidbody GameObjects of the joint to prevent the anchor position updating their position
```

Comment:
```suggestion
            //Disable any child rigidbody GameObjects of the joint to prevent the anchor position updating their position.
```
Just because you did the same for the other comments. One day we'll get conventions documented for comments, I for example would also prefix by a space like `// Disable ...` for example 😄